### DISCUSSION TITLE
Set "target=_blank" for external urls in child theme?

### DISCUSSION THREAD

**Discussion by iamsantanubanerjee** [thuliteio/doks#588] [Q&A] [answered]
Is there a way to make the external links open in a different tab automatically?
Like, social links in the nav bar?

Thanks
Santanu Banerjee

**Comment by h-enk** [comment, marked as answer]
I advise you to open external links in a new window or tab only when necessary — for UX reasons mainly, see also [G200: Opening new windows and tabs from a link only when necessary](https://www.w3.org/TR/WCAG20-TECHS/G200.html)

For all external links in the content, you could create a file `./layouts/_default/_markup/render-link.html`, and add:

```html
<a href="{{ .Destination | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}{{ if strings.HasPrefix .Destination "http" }} target="_blank" rel="noopener"{{ end }}>{{ .Text | safeHTML }}</a>
```

See also: 
- [Attributes - target](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target)
- [Link with title Markdown example](https://gohugo.io/getting-started/configuration-markup/#link-with-title-markdown-example)

For links in menus, you could update the corresponding layout partial, e.g. change `./layouts/partials/header/header.html` (line 46) to:

```html
<a target="_blank" rel="noopener" class="nav-link ps-0 py-1" href="{{ .URL | relURL }}">{{ .Pre | safeHTML }}<small class="ms-2 d-md-none">{{ .Name | safeHTML }}</small></a>
```
